### ISSUE TITLE
Split vignette in "for users" and "for developers"?

### ISSUE THREAD

**Issue by Bisaloo** [futureverse/progressr#90]
Hi,

I think it'd be super useful to have a vignette that is targeted at end users of packages relying on progressr. I want to direct users of my packages to a short explanation about how that they can enable progress report if desired, and customize the type of report. However, the current vignette also provides information about the philosophy of the package, and information to package developers, which may be not relevant, and could overthrow a user that just wants a simple explanation on how to get a progress bar.

The [drat package](https://cran.r-project.org/package=drat) takes this approach of offering two vignettes and I find it very helpful.

I've thought about putting this introduction in my own package but I think it makes more sense if it's within progressr since it'd be useful for many other packages depending on progressr.

**Comment by HenrikBengtsson** [comment]
Yes, this is a great suggestion.  The reason for the current, rather long, vignette/README is that it grew out of my own personal note when I prototyped this package (which was only meant to be a proof-of-concept package).   There are new things planned for the package (e.g. global calling handlers) that will further affect "best practices for developers" and "best practices for users".  When that is in place, I'll revisit the vignette structure.

**Comment by HenrikBengtsson** [comment]
I've now split up the rather long mono-vignette into eight separate vignettes. See 'Articles' menu on https://progressr.futureverse.org/index.html. Things might be refined, but that's the start.
